hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -59,6 +59,21 @@
 except ImportError:
     _GenericAlias = ()
 
+try:
+    from typing_extensions import _AnnotatedAlias  # type: ignore
+except ImportError:
+    try:
+        from typing_extensions import AnnotatedMeta as _AnnotatedAlias  # type: ignore
+
+        assert sys.version_info[:2] == (3, 6)
+    except ImportError:
+        _AnnotatedAlias = ()  # type: ignore
+
+
+def is_annotated_instance(thing):
+    # TODO: this should handle `typing.Annotated` too, for when `typing_extensions` is not installed.
+    return isinstance(thing, _AnnotatedAlias)
```

Comment:
Let's inline this logic into `is_annotated_type()`.  You'll also need to actually do the `typing.Annotated` handling; that's why the tests are currently failing on py39 and py310.